src/pages/Dashboard/Modals/SupplyWithdraw/SupplyWithdrawForm.tsx

```diff
@@ -0,0 +1,201 @@
+/** @jsxImportSource @emotion/react */
+import React, { useState, useEffect } from 'react';
+import BigNumber from 'bignumber.js';
+import { FormikProps } from 'formik';
+import { Typography } from '@mui/material';
+import { FormValues } from 'containers/AmountForm/validationSchema';
+import { IAmountFormProps } from 'containers/AmountForm';
+import {
+  TokenTextField,
+  Delimiter,
+  InfoRow,
+  IInfoRowProps,
+  PrimaryButton,
+  ProgressBarHorizontal,
+  Tooltip,
+  ValueUpdate,
+} from 'components';
+import { useTranslation } from 'translation';
+import { Asset, TokenId } from 'types';
+import {
+  getBigNumber,
+  convertCoinsToWei,
+  convertWeiToCoins,
+  formatCentsToReadableValue,
+  format,
+  currencyFormatter,
+} from 'utilities/common';
+import { useStyles } from '../styles';
+
+interface ISupplyWithdrawFormProps extends FormikProps<FormValues> {
+  asset: Asset;
+  tokenInfo: IInfoRowProps['info'];
+  userTotalBorrowBalance: BigNumber;
+  userTotalBorrowLimit: BigNumber;
+  dailyEarnings: BigNumber;
+  onSubmit: IAmountFormProps['onSubmit'];
+  enabledButtonKey: string;
+  disabledButtonKey: string;
+  maxInput: BigNumber;
+  newBalance: (amount: BigNumber) => BigNumber;
+}
+
+export const SupplyWithdrawForm: React.FC<ISupplyWithdrawFormProps> = ({
+  values,
+  setFieldValue,
+  resetForm,
+  asset,
+  tokenInfo,
+  userTotalBorrowBalance,
+  userTotalBorrowLimit,
+  dailyEarnings,
+  maxInput,
+  enabledButtonKey,
+  disabledButtonKey,
+  newBalance,
+}) => {
+  const { id: assetId } = asset;
+  const { amount } = values;
+  const validAmount = amount && !amount.isZero() && !amount.isNaN();
+  const styles = useStyles();
+  const { t, Trans } = useTranslation();
+  const [newBorrowLimit, setNewBorrowLimit] = useState<BigNumber | undefined>(undefined);
+  const [newBorrowPercent, setNewBorrowPercent] = useState(
+    userTotalBorrowBalance.div(userTotalBorrowLimit).times(100),
+  );
+  useEffect(() => resetForm, []);
+  useEffect(() => {
+    const tokenPrice = getBigNumber(asset?.tokenPrice);
+    const collateralFactor = getBigNumber(asset?.collateralFactor);
+
+    if (tokenPrice && validAmount) {
+      const amountInUsd = convertWeiToCoins({
+        value: amount as BigNumber,
+        tokenSymbol: asset.id,
+      })
+        .times(tokenPrice)
+        .times(collateralFactor);
+      const temp = newBalance(amountInUsd);
+      setNewBorrowLimit(BigNumber.maximum(temp, 0));
+      setNewBorrowPercent(userTotalBorrowBalance.div(temp).times(100));
+    } else if (BigNumber.isBigNumber(userTotalBorrowLimit)) {
+      setNewBorrowLimit(undefined);
+      setNewBorrowPercent(userTotalBorrowBalance.div(userTotalBorrowLimit).times(100));
+    }
+  }, [amount, asset?.id, userTotalBorrowBalance, userTotalBorrowLimit]);
```

Comment:
`useMemo` is not a replacement for `useState`. `useMemo` is a performance optimization while `useState` is to guarantee updates. I believe `useState` is the correct choice here and `useMemo` could be used to memoize values being set in state so they aren't calculated on every render.

```
You may rely on useMemo as a performance optimization, not as a semantic guarantee. In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without useMemo — and then add it to optimize performance.
```
[source](https://reactjs.org/docs/hooks-reference.html#usememo)